packages/main/src/plugin/kubernetes/kubernetes-client.spec.ts

```diff
@@ -265,6 +270,17 @@ const apiSender: ApiSenderType = {
   receive: vi.fn(),
 };
 
+const mocks = vi.hoisted(() => ({
+  getServiceMock: vi.fn(),
+}));
+
+vi.mock('/@/plugin/kubernetes/kubernetes-port-forward-service', () => ({
+  KubernetesPortForwardService: vi.fn(),
+  KubernetesPortForwardServiceProvider: vi.fn().mockReturnValue({
+    getService: mocks.getServiceMock,
+  } as unknown as KubernetesPortForwardServiceProvider),
+}));
```

Comment:
I don't like having to use `vi.hoisted` but I do not have any choice here...

The `KubernetesPortForwardServiceProvider` is not initialized in the constructor of the KubeClient but statically, when the file is imported

https://github.com/podman-desktop/podman-desktop/blob/36d8254e73baafcc79e9a3640e92772e8aaae89a/packages/main/src/plugin/kubernetes/kubernetes-client.ts#L182

This mean the `KubernetesPortForwardServiceProvider` constructor is called before the beforeEach and before the tests are ran... so we cannot mock in other places...